campaigns.py

```diff
@@ -6,52 +6,86 @@
 
 from jinja2 import Template
 
-import vm
+import app
+
+
+env['eagerly_disconnect'] = True
+
+
+APPLICATIONS = ['frontend', 'static']
+CAMPAIGN_CLASSES = ['red', 'black', 'green']
+
 
 def validate_classes(campaign_class):
     """Checks that the campaign class is valid"""
-    if campaign_class in ['red', 'black', 'green']:
+    if campaign_class in CAMPAIGN_CLASSES:
         return campaign_class
     raise Exception, "Invalid class %s, valid values are 'red', 'black', 'green'" % campaign_class
 
 
 @runs_once
-def homepage_template():
-    """Promotes a campaign to the homepage of GOV.UK"""
-    context = {
-        'heading': prompt("Heading for campaign:"),
-        'extra_info': prompt("Extra information for campaign:"),
-        'more_info': prompt("Link for more information:"),
-        'campaign_class': prompt("Campaign class:", validate=validate_classes)
+def set_context():
+    env['context'] = {
+        'heading': prompt("Heading for campaign:", key='heading'),
+        'extra_info': prompt("Extra information for campaign:", key='extra_info'),
+        'more_info': prompt("Link for more information:", key='more_info'),
+        'campaign_class': prompt("Campaign class:", key='campaign_class', validate=validate_classes)
     }
 
-    template = Template("""<div id="campaign" class="{{ campaign_class }}">
-  <div class="campaign-inner">
-    <h1>{{ heading|e }}</h1>
-    <p>{{ extra_info|e }}</p>
-    <a href="{{ more_info|e }}">More information</a>
-  </div>
-</div>""")
-    env['template_contents'] = template.render(context)
 
-    print "Template contents:\n%s" % env['template_contents']
+def template(app):
+    if app == 'frontend':
+      template = Template("""<div id="campaign" class="{{ campaign_class }}">
+    <div class="campaign-inner">
+      <h1>{{ heading|e }}</h1>
+      <p>{{ extra_info|e }}</p>
+      <a href="{{ more_info|e }}">More information</a>
+    </div>
+  </div>""")
+    elif app == 'static':
+      template = Template("""<p>{{ heading|e }}<br />
+    {{ extra_info|e }}</p>
+  <a href="#" class="right">{{ more_info|e }}</a>""")
+
+    env['template_contents'] = template.render(env.context)
+
 
 @task
 @roles('class-frontend')
-def deploy_to_homepage():
-    execute(homepage_template)
+def banner(application, action):
+    if action == 'deploy':
```

Comment:
There's a lot of `if` logic here. I think it would be cleaner to move each action into its own method. It would also make the intent cleaner; calling `banner` doesn't describe well what action is being invoked.